lib/internal/util.js

```diff
@@ -314,7 +320,7 @@ function promisify(original) {
   function fn(...args) {
     const promise = createPromise();
     try {
-      original.call(this, ...args, (err, ...values) => {
+      original.call(context || this, ...args, (err, ...values) => {
```

Comment:
what if a non-object is passed (like the number 0)? Please check explicitly (and add a test for it) :)